### PR TITLE
Update FFI chapters to reflect RFC 68

### PR DESCRIPTION
**Pull request by ergl**
Changes related to [RFC 68](https://github.com/ponylang/rfcs/blob/main/text/0068-force-declaration-of-ffi-functions.md)

### CODE REVIEW

**Comment by ergl** [comment]
This is ready for review

File: docs/c-ffi/calling-c.md
Comment (jemc) [inline docs/c-ffi/calling-c.md:104]: Small suggestion to make it a bit more clear how creepy and subtle this problem can be:

```suggestion
In the example above, the compiler will type-check the first argument to `printf`, but will not be able to check any other argument, since it lacks the necessary type information. It is __very__ important that you use `...` in the FFI signature if the corresponding C function is variadic: if you don't, the compiler might generate a program that is incorrect or crash on some platforms while appearing to work correctly on others.
```

**Review by jemc — APPROVED** [review]

**Review by jemc — APPROVED** [review]

**Comment by ergl** [comment]
@jemc I'm going to add some clarifying comments about what "compatible" type declarations really mean.

**Comment by SeanTAllen** [comment]
@ergl does that mean i shouldnt merge this yet?

**Comment by ergl** [comment]
@SeanTAllen No, not yet, I want to give others a chance to review the latest additions.

**Comment by SeanTAllen** [comment]
@ergl who else do you want to review?

**Comment by ergl** [comment]
@SeanTAllen if either you or @jemc or you can give a look at the latest section I've added, that would be great.

File: docs/appendices/examples.md
Comment (SeanTAllen) [inline docs/appendices/examples.md:77]: is this an intentional removal @ergl ?

File: docs/c-ffi/calling-c.md
Comment (SeanTAllen) [inline docs/c-ffi/calling-c.md:16]: whats the reason for the removal of this? At least out of context, this seems like important information.

File: docs/appendices/examples.md
Comment (ergl) [reply docs/appendices/examples.md:77]: Yes, this specific part is already covered here https://tutorial.ponylang.io/c-ffi/calling-c.html#get-and-pass-pointers-to-ffi, so it feels like a duplicate

File: docs/c-ffi/calling-c.md
Comment (ergl) [reply docs/c-ffi/calling-c.md:16]: My rationale is that by removing the syntax for writing out the return type of a FFI function at the call site and requiring a declaration, it's hard to show a one-line example, so I decided to skip the intro and start the chapter by showing how to write a declaration.

The paragraph about having to type `U64(1)` is not needed anymore, since the compiler can now infer the correct type by looking at the declaration. This means one is allowed to write `1` as the second argument without any problems.

The bit about the @ symbol is covered a bit later.

File: docs/appendices/examples.md
Comment (SeanTAllen) [reply docs/appendices/examples.md:77]: Gotcha. Looks reasonable.

File: docs/c-ffi/calling-c.md
Comment (SeanTAllen) [reply docs/c-ffi/calling-c.md:16]: Ok, I am ok with this. We can see if people have any issues with it.

**Review by SeanTAllen — APPROVED** [review]

**Comment by SeanTAllen** [comment]
@ergl do you want @jemc to give another review as well?

**Comment by SeanTAllen** [comment]
I added a DO NOT MERGE label as this shouldn't be merged until the corresponding changes to ponyc are released. I'll be doing the merging as part of the release process for the next ponyc.

**Comment by ergl** [comment]
@SeanTAllen If he has the time to review the last commit, yes, but it's not a problem otherwise.

**Review by jemc — APPROVED** [review]

**Pull request merged**
